doc/read-the-docs-site/adr/0006-indexer-resuming-strategy.rst

```diff
@@ -45,16 +45,89 @@ Decision
   In the current Marconi interface, it is actually just ``securityParam + 1`` as we assume that all rollbackable blocks *can* be fully stored on disk because of the disk flush.
   We assume that all the data contained in the indexer is consistent for all the point returned by an indexer.
 
-* We will have each indexer start the node-to-client chain-sync protocol at different points in time.
-  However, the ``Coordinator`` will make sure that all indexers are synced until a common point, and advance at the same speed (i.e. they can only request the next block once all the indexers have finished processing the current block).
+* We will start a single node chain-sync client for all indexers instead of one node chain sync client per indexer.
 
-* We will use the full set of points provided by ``resumableFromStorage`` of an indexer as resuming points for the node-to-client chain-sync protocol.
-  These points need to ordered in descending order so that protocol can priotise the selection of the latest ones.
+* We will implement the following resuming logic.
+
+  In order to start the chain-sync client, we need to provide to provide intersection points (or resuming points).
+  Intersection points represent the points from which we want to start syncing from the local node.
+  We will provide as intersection points the points returned from the following pseudo-code:
+
+  .. code-block:: haskell
+
+    -- | Query the resumable points of each indexer then select the resumable
+    -- points of the indexer that has the lowest immutable point. If two indexer
+    -- have the same lowest immutable point, select the one which has the lowest
+    -- overall point (immutable or not).
+    selectIntersectionPoints indexers = do
+      resumingPoints <- mapM queryResumingPoints indexers
+      head $ sortOn (\rps -> (maximum $ filter isImmutablePoint rps, maximum rps)) resumingPoints
+
+  Next, we need a function which processes each ``ChainSyncEvent`` returned by the chain-sync client for each indexer.
+  The pseudo-code is as follows:
+
+  .. code-block:: haskell
+
+    -- Preconditions:
+    --   * The 'initialLastSyncPoints' param should be sorted in ascending order on SlotNo
+    --   * Each element of the 'initialLastSyncPoints' param should satisfy the property 'getSlotNo point == getSlotNo (point + 1) + 1'.
```

Comment:
It sounds unclear to me (and probably wrong, as it implies `getSlotNo point - 1 = getSlotNo (point + 1)` and thus `slotNo`s decrease when we move to a next point)

Reply:
Did you see the `+1` after `(point + 1)`?

Reply:
yes and I computed `getSlotNo point == getSlotNo (point + 1) + 1  ~~> getSlotNo point - 1 == getSlotNo (point + 1)`

Reply:
You're totally right :+1: 

I should be defined in terms of blockNo then. `getBlockNo point - 1 == getBlockNo (succ point)`

Reply:
Now I get it :)